### DISCUSSION TITLE
Welcome to kube-burner Discussions!

### DISCUSSION THREAD

**Discussion by smalleni** [kube-burner/kube-burner#512] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ritankarsaha** [comment]
Hey I was just wondering if there are dev meets or discussions for this project ?

**Reply by rsevilla87** [reply]
Correct, we have a bi-weekly meeting

**Reply by rsevilla87** [reply]
https://zoom-lfx.platform.linuxfoundation.org/meeting/93448055215?password=a34445ac-64af-4e01-9446-3864dff5a243

**Reply by ritankarsaha** [reply]
ohh sure @rsevilla87 !! Thanks for sharing.  😁

**Comment by ritankarsaha** [comment]
Also I had another doubt, do we need to run pre-commit only for the files we have changes or modified and check if linting errors exist for that ? 
On running `make lint` or basically `pre-commit run --all-files`  after just a clone with no file changed, errors are thrown for some files

```
Executing pre-commit for all files
pre-commit run --all-files
[WARNING] Unexpected key(s) present at root: version
golangci-lint............................................................Failed
- hook id: golangci-lint
- exit code: 1

pkg/watchers/watcher_manager.go:55:19: ST1005: error strings should not be capitalized (staticcheck)
                        wm.recordError(fmt.Errorf("Error getting REST client for %s: %w", kindLower, err))
                                       ^
pkg/watchers/watcher_manager.go:70:19: ST1005: error strings should not be capitalized (staticcheck)
                        wm.recordError(fmt.Errorf("Error starting %s: %w", watcherName, err))
                                       ^
2 issues:
* staticcheck: 2

markdownlint.............................................................Passed
Test shell scripts with shellcheck.......................................Passed
check json...............................................................Passed
make: *** [Makefile:64: lint] Error 1
```

This was happening after I made some changes for an issue before opening a PR, so checked if the same linting errors are there for the main-branch as well. 
Could you please clarify this once ?  
If pre-commit is to run for all the changed files only, then probably my implementation is correct

@smalleni @chentex

**Reply by ygalblum** [reply]
The linter will check the entire code, not only your change. Because it runs as part of the CI, the `main` branch should be error  free. 
The reason you got this is a hiccup where we updated the linter version (which now complains about this) just before `watcher_manager.go` was added. The latter was added without running the test after rebasing to main. 
I've opened a PR to fix these errors: https://github.com/kube-burner/kube-burner/pull/909. For now you can just replace `Error` with `error` and once the PR merged, you will be able to rebase on `main` and be free of this

**Reply by ritankarsaha** [reply]
> The linter will check the entire code, not only your change. Because it runs as part of the CI, the `main` branch should be error  free. 
> 
> The reason you got this is a hiccup where we updated the linter version (which now complains about this) just before `watcher_manager.go` was added. The latter was added without running the test after rebasing to main. 
> 
> I've opened a PR to fix these errors: https://github.com/kube-burner/kube-burner/pull/909. For now you can just replace `Error` with `error` and once the PR merged, you will be able to rebase on `main` and be free of this 

Ohh sure thenn !! 
I will be pushing the changes with the fix as of now. 
Thanks for clarifyingg !! @ygalblum

**Comment by ritankarsaha** [comment]
Hi,
The `test-k8s.bats` test suite is consistently failing during setup due to KubeVirt timeouts:

The error messages thrown is quite like this -- 

```bash
error: timed out waiting for the condition on kubevirts/kubevirt
```

Probable Issue:

- Kind cluster creates successfully
- KubeVirt operator deploys but times out waiting for kubevirts/kubevirt to become available
This causes setup_file to fail, preventing any actual tests from running
Affects both `make test` and `make test-k8s` commands.
This failure is happening on the `main` branch itself with no changes made. 

Environment:
Ubuntu 22.04 LTS
kind version 0.22.0
Consistent failure across multiple runs

However Tests pass when run on a pre-existing cluster without KubeVirt.

Is this a known issue? Or is facing this unusual? Do we need to increase KubeVirt wait timeout in helpers.bash or any turnaround ?

**Reply by rsevilla87** [reply]
Can you open an issue to track this properly?

Thanks

**Reply by ritankarsaha** [reply]
Sure @rsevilla87 also faced a similar type of issue ig in my PR #914 . On investigating the CI errors they mainly show missing metrics files, Prometheus query errors, or unrelated timeouts and not related with the code changes I made, i guess.

**Reply by ritankarsaha** [reply]
Hey @rsevilla87 created the issue for the same at #915 . Is this a known issue? Does it run properly on others' systems?
